Failure to establish a push connection in Vaadin 8.0.0

After upgrading to Vaadin 8.0.0, the browser failed to establish a WebSocket connection with the Server Push. This is the log from Chrome's console:
```
'ws://localhost:8080/vaadinServlet/PUSH?v-uiId=0&v-csrfToken=5a8ef0f3-7e53-4bbe-b3c0-126a7de1cce0&X-Atmosphere-tracking-id=0&X-Atmosphere-Framework=2.3.2.vaadin1-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json;%20charset=UTF-8&X-atmo-protocol=true'

failed: Error during WebSocket handshake: Unexpected response code: 200
```

I think there is a version mismatch in the Atmosphere framework. In the error above it uses `2.3.2` but in my console and running in DEBUG_MODE, it shows `2.4.5`:

```
06:20:22.239 [http-nio-8080-exec-2] INFO  o.atmosphere.cpr.AtmosphereFramework - Atmosphere Framework 2.4.5.vaadin1 started.
06:20:22.244 [http-nio-8080-exec-2] INFO  o.atmosphere.cpr.AtmosphereFramework - Installed AtmosphereInterceptor  Track Message Size Interceptor using | with priority BEFORE_DEFAULT 
06:20:27.021 [http-nio-8080-exec-3] INFO  o.a.coyote.http11.Http11Processor - Error parsing HTTP request header
 Note: further occurrences of HTTP header parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in the request target. The valid characters are defined in RFC 7230 and RFC 3986
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:471)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:667)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:798)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1434)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

Could this be the reason behind it?

I'm facing this issue too, I am using Spring Boot 1.5.2 with Apache Tomcat/8.5.11

`07/03/17 19:59:44.718 [ WARN][Atmosphere-Shared-0][JDK14LoggerAdapter:588][log] : Exception during suspend() operation java.lang.NullPointerException
07/03/17 19:59:44.737 [ERROR][Atmosphere-Shared-0][PushAtmosphereHandler$AtmosphereResourceListener:114][onThrowable] : Exception in push connection
java.lang.NullPointerException: null
	at org.apache.catalina.connector.Request.getServletContext(Request.java:1627) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.Request.getContextPath(Request.java:1984) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.RequestFacade.getContextPath(RequestFacade.java:784) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at javax.servlet.http.HttpServletRequestWrapper.getContextPath(HttpServletRequestWrapper.java:150) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.springframework.web.util.UrlPathHelper.getContextPath(UrlPathHelper.java:325) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinApplication(UrlPathHelper.java:224) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinServletMapping(UrlPathHelper.java:177) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at com.vaadin.spring.server.SpringVaadinServletRequest.getPathInfo(SpringVaadinServletRequest.java:68) ~[vaadin-spring-2.0.0.jar:2.0.0]
	at org.atmosphere.cpr.AtmosphereRequestImpl.getPathInfo(AtmosphereRequestImpl.java:162) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at javax.servlet.http.HttpServletRequestWrapper.getPathInfo(HttpServletRequestWrapper.java:132) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at com.vaadin.server.ServletPortletHelper.hasPathPrefix(ServletPortletHelper.java:90) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.ServletPortletHelper.isAppRequest(ServletPortletHelper.java:139) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinServletService.isOtherRequest(VaadinServletService.java:168) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinServletService.requestCanCreateSession(VaadinServletService.java:155) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinService.findOrCreateVaadinSession(VaadinService.java:703) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinService.findVaadinSession(VaadinService.java:572) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.communication.PushHandler.connectionLost(PushHandler.java:318) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.communication.PushAtmosphereHandler$AtmosphereResourceListener.onDisconnect(PushAtmosphereHandler.java:109) ~[vaadin-server-8.0.1.jar:8.0.1]
	at org.atmosphere.cpr.AtmosphereResourceImpl.onDisconnect(AtmosphereResourceImpl.java:754) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResourceImpl.notifyListeners(AtmosphereResourceImpl.java:645) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResponseImpl.handleException(AtmosphereResponseImpl.java:729) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResponseImpl.write(AtmosphereResponseImpl.java:810) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResponseImpl.write(AtmosphereResponseImpl.java:783) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.writeReady(AtmosphereInterceptorWriter.java:92) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.invokeInterceptor(AtmosphereInterceptorWriter.java:83) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.write(AtmosphereInterceptorWriter.java:69) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.write(AtmosphereInterceptorWriter.java:64) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResponseImpl$Stream.write(AtmosphereResponseImpl.java:956) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.handler.AbstractReflectorAtmosphereHandler.onStateChange(AbstractReflectorAtmosphereHandler.java:155) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at com.vaadin.server.communication.PushAtmosphereHandler.onStateChange(PushAtmosphereHandler.java:52) [vaadin-server-8.0.1.jar:8.0.1]
	at org.atmosphere.cpr.DefaultBroadcaster.invokeOnStateChange(DefaultBroadcaster.java:1047) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.prepareInvokeOnStateChange(DefaultBroadcaster.java:1067) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.executeAsyncWrite(DefaultBroadcaster.java:872) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster$2.run(DefaultBroadcaster.java:474) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
07/03/17 19:59:44.744 [ERROR][Atmosphere-Shared-0][PushAtmosphereHandler$AtmosphereResourceListener:114][onThrowable] : Exception in push connection
java.lang.RuntimeException: java.lang.NullPointerException
	at org.atmosphere.cpr.AtmosphereResponseImpl.write(AtmosphereResponseImpl.java:811) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResponseImpl.write(AtmosphereResponseImpl.java:783) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.writeReady(AtmosphereInterceptorWriter.java:92) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.invokeInterceptor(AtmosphereInterceptorWriter.java:83) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.write(AtmosphereInterceptorWriter.java:69) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereInterceptorWriter.write(AtmosphereInterceptorWriter.java:64) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResponseImpl$Stream.write(AtmosphereResponseImpl.java:956) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.handler.AbstractReflectorAtmosphereHandler.onStateChange(AbstractReflectorAtmosphereHandler.java:155) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at com.vaadin.server.communication.PushAtmosphereHandler.onStateChange(PushAtmosphereHandler.java:52) ~[vaadin-server-8.0.1.jar:8.0.1]
	at org.atmosphere.cpr.DefaultBroadcaster.invokeOnStateChange(DefaultBroadcaster.java:1047) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.prepareInvokeOnStateChange(DefaultBroadcaster.java:1067) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.executeAsyncWrite(DefaultBroadcaster.java:872) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster$2.run(DefaultBroadcaster.java:474) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
Caused by: java.lang.NullPointerException: null
	at org.apache.coyote.http11.Http11OutputBuffer.commit(Http11OutputBuffer.java:351) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.coyote.http11.Http11Processor.prepareResponse(Http11Processor.java:1288) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.coyote.AbstractProcessor.action(AbstractProcessor.java:254) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.coyote.Response.action(Response.java:168) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.coyote.http11.Http11OutputBuffer.doWrite(Http11OutputBuffer.java:217) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.coyote.Response.doWrite(Response.java:517) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.OutputBuffer.realWriteBytes(OutputBuffer.java:351) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.OutputBuffer.appendByteArray(OutputBuffer.java:778) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.OutputBuffer.append(OutputBuffer.java:707) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.OutputBuffer.writeBytes(OutputBuffer.java:391) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.OutputBuffer.write(OutputBuffer.java:369) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.CoyoteOutputStream.write(CoyoteOutputStream.java:96) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.CoyoteOutputStream.write(CoyoteOutputStream.java:89) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.atmosphere.cpr.AtmosphereResponseImpl.write(AtmosphereResponseImpl.java:802) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	... 17 common frames omitted
07/03/17 19:59:44.745 [ WARN][Atmosphere-Shared-0][JDK14LoggerAdapter:588][log] : Exception during suspend() operation java.lang.NullPointerException
07/03/17 19:59:44.748 [ERROR][Atmosphere-Shared-0][PushAtmosphereHandler$AtmosphereResourceListener:114][onThrowable] : Exception in push connection
java.lang.NullPointerException: null
	at org.apache.catalina.connector.Request.getServletContext(Request.java:1627) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.Request.getContextPath(Request.java:1984) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.RequestFacade.getContextPath(RequestFacade.java:784) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at javax.servlet.http.HttpServletRequestWrapper.getContextPath(HttpServletRequestWrapper.java:150) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.springframework.web.util.UrlPathHelper.getContextPath(UrlPathHelper.java:325) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinApplication(UrlPathHelper.java:224) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinServletMapping(UrlPathHelper.java:177) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at com.vaadin.spring.server.SpringVaadinServletRequest.getPathInfo(SpringVaadinServletRequest.java:68) ~[vaadin-spring-2.0.0.jar:2.0.0]
	at org.atmosphere.cpr.AtmosphereRequestImpl.getPathInfo(AtmosphereRequestImpl.java:162) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at javax.servlet.http.HttpServletRequestWrapper.getPathInfo(HttpServletRequestWrapper.java:132) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at com.vaadin.server.ServletPortletHelper.hasPathPrefix(ServletPortletHelper.java:90) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.ServletPortletHelper.isAppRequest(ServletPortletHelper.java:139) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinServletService.isOtherRequest(VaadinServletService.java:168) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinServletService.requestCanCreateSession(VaadinServletService.java:155) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinService.findOrCreateVaadinSession(VaadinService.java:703) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinService.findVaadinSession(VaadinService.java:572) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.communication.PushHandler.connectionLost(PushHandler.java:318) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.communication.PushAtmosphereHandler$AtmosphereResourceListener.onThrowable(PushAtmosphereHandler.java:116) ~[vaadin-server-8.0.1.jar:8.0.1]
	at org.atmosphere.cpr.AtmosphereResourceImpl.onThrowable(AtmosphereResourceImpl.java:704) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResourceImpl.notifyListeners(AtmosphereResourceImpl.java:654) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.onException(DefaultBroadcaster.java:1130) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.onException(DefaultBroadcaster.java:1112) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.invokeOnStateChange(DefaultBroadcaster.java:1050) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.prepareInvokeOnStateChange(DefaultBroadcaster.java:1067) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.executeAsyncWrite(DefaultBroadcaster.java:872) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster$2.run(DefaultBroadcaster.java:474) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
07/03/17 19:59:44.749 [ WARN][Atmosphere-Shared-0][JDK14LoggerAdapter:588][log] : Listener error {}
java.lang.NullPointerException: null
	at org.apache.catalina.connector.Request.getServletContext(Request.java:1627) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.Request.getContextPath(Request.java:1984) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.apache.catalina.connector.RequestFacade.getContextPath(RequestFacade.java:784) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at javax.servlet.http.HttpServletRequestWrapper.getContextPath(HttpServletRequestWrapper.java:150) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at org.springframework.web.util.UrlPathHelper.getContextPath(UrlPathHelper.java:325) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinApplication(UrlPathHelper.java:224) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinServletMapping(UrlPathHelper.java:177) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at com.vaadin.spring.server.SpringVaadinServletRequest.getPathInfo(SpringVaadinServletRequest.java:68) ~[vaadin-spring-2.0.0.jar:2.0.0]
	at org.atmosphere.cpr.AtmosphereRequestImpl.getPathInfo(AtmosphereRequestImpl.java:162) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at javax.servlet.http.HttpServletRequestWrapper.getPathInfo(HttpServletRequestWrapper.java:132) ~[tomcat-embed-core-8.5.11.jar:8.5.11]
	at com.vaadin.server.ServletPortletHelper.hasPathPrefix(ServletPortletHelper.java:90) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.ServletPortletHelper.isAppRequest(ServletPortletHelper.java:139) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinServletService.isOtherRequest(VaadinServletService.java:168) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinServletService.requestCanCreateSession(VaadinServletService.java:155) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinService.findOrCreateVaadinSession(VaadinService.java:703) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.VaadinService.findVaadinSession(VaadinService.java:572) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.communication.PushHandler.connectionLost(PushHandler.java:318) ~[vaadin-server-8.0.1.jar:8.0.1]
	at com.vaadin.server.communication.PushAtmosphereHandler$AtmosphereResourceListener.onThrowable(PushAtmosphereHandler.java:116) ~[vaadin-server-8.0.1.jar:8.0.1]
	at org.atmosphere.cpr.AtmosphereResourceImpl.onThrowable(AtmosphereResourceImpl.java:704) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.AtmosphereResourceImpl.notifyListeners(AtmosphereResourceImpl.java:673) ~[atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.onException(DefaultBroadcaster.java:1130) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.onException(DefaultBroadcaster.java:1112) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.invokeOnStateChange(DefaultBroadcaster.java:1050) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.prepareInvokeOnStateChange(DefaultBroadcaster.java:1067) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster.executeAsyncWrite(DefaultBroadcaster.java:872) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at org.atmosphere.cpr.DefaultBroadcaster$2.run(DefaultBroadcaster.java:474) [atmosphere-runtime-2.4.5.vaadin1.jar:2.4.5.vaadin1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
`

--- a/pom.xml
+++ b/pom.xml
@@ -34,7 +34,7 @@
         <javax.portlet.version>2.0</javax.portlet.version>
         <vaadin.sass.version>0.9.13</vaadin.sass.version>
         <!-- Note that this should be kept in sync with the class Constants -->
-        <atmosphere.runtime.version>2.4.5.vaadin1</atmosphere.runtime.version>
+        <atmosphere.runtime.version>2.4.5.vaadin2</atmosphere.runtime.version>
 
         <!-- OSGi -->
         <osgi.javax.servlet.version>3.0.0</osgi.javax.servlet.version>

--- a/server/src/main/java/com/vaadin/server/Constants.java
+++ b/server/src/main/java/com/vaadin/server/Constants.java
@@ -66,7 +66,7 @@ public interface Constants {
             + "=================================================================";
 
     // Keep the version number in sync with pom.xml
-    static final String REQUIRED_ATMOSPHERE_RUNTIME_VERSION = "2.4.5.vaadin1";
+    static final String REQUIRED_ATMOSPHERE_RUNTIME_VERSION = "2.4.5.vaadin2";
 
     static final String INVALID_ATMOSPHERE_VERSION_WARNING = "\n"
             + "=================================================================\n"

